### PR TITLE
NMS-16956: Added PTP poller and tests

### PR DESCRIPTION
**Pull request by christianpape**
* JIRA: https://opennms.atlassian.net/browse/NMS-16956

### CODE REVIEW

File: opennms-base-assembly/src/main/filtered/etc/poller-configuration.xml [hunk 1]
```diff
@@ -213,6 +213,13 @@
          <parameter key="rrd-base-name" value="pop3" />
          <parameter key="ds-name" value="pop3" />
       </service>
+      <service name="PTP" interval="300000" user-defined="false" status="on">
+         <pattern><![CDATA[^PTP-(?<clockPort>.*)-(?<portState>.*)$]]></pattern>
+         <parameter key="rrd-repository" value="${install.share.dir}/rrd/response" />
+         <parameter key="rrd-base-name" value="ptp" />
```
Comment (indigo423) [inline opennms-base-assembly/src/main/filtered/etc/poller-configuration.xml:219]: See comment in #7539

File: opennms-base-assembly/src/main/filtered/etc/poller-configuration.xml [hunk 2]
```diff
@@ -213,6 +213,13 @@
          <parameter key="rrd-base-name" value="pop3" />
          <parameter key="ds-name" value="pop3" />
       </service>
+      <service name="PTP" interval="300000" user-defined="false" status="on">
```
Comment (indigo423) [inline opennms-base-assembly/src/main/filtered/etc/poller-configuration.xml:216]: See comment in #7539

File: opennms-base-assembly/src/main/filtered/etc/datacollection/ptpbase.xml [hunk 3]
```diff
@@ -0,0 +1,218 @@
...
+   </resourceType>
+   <resourceType name="ptpbaseClockPortAssociateEntry" label="ptpbaseClockPortAssociateEntry" resourceLabel="${index}">
+      <persistenceSelectorStrategy class="org.opennms.netmgt.collection.support.PersistAllSelectorStrategy"/>
+      <storageStrategy class="org.opennms.netmgt.collection.support.IndexStorageStrategy"/>
+   </resourceType>
+   <group name="ptpbaseSystemTable" ifType="all">
+      <mibObj oid=".1.3.6.1.2.1.241.1.1.1.1.1" instance="ptpbaseSystemEntry" alias="ptpDomainIndex" type="integer"/>
+      <mibObj oid=".1.3.6.1.2.1.241.1.1.1.1.2" instance="ptpbaseSystemEntry" alias="ptpInstanceIndex" type="integer"/>
+      <mibObj oid=".1.3.6.1.2.1.241.1.1.1.1.3" instance="ptpbaseSystemEntry" alias="ptpDomaClocPortTota" type="gauge32"/>
+   </group>
```
Comment (indigo423) [inline opennms-base-assembly/src/main/filtered/etc/datacollection/ptpbase.xml:58]: We have started to separate the resource types in `resource-types.d`, we might want to do the same for these here as well.

File: opennms-base-assembly/src/main/filtered/etc/datacollection/ptpbase.xml [hunk 4]
```diff
@@ -0,0 +1,218 @@
...
+      <mibObj oid=".1.3.6.1.2.1.241.1.2.11.1.6" instance="ptpbaseClockPortAssociateEntry" alias="ptpCloPorAssAddType" type="string"/>
+      <mibObj oid=".1.3.6.1.2.1.241.1.2.11.1.7" instance="ptpbaseClockPortAssociateEntry" alias="ptpClocPortAssoAddr" type="octetstring"/>
+      <mibObj oid=".1.3.6.1.2.1.241.1.2.11.1.8" instance="ptpbaseClockPortAssociateEntry" alias="ptpCloPorAssPacSent" type="counter64"/>
+      <mibObj oid=".1.3.6.1.2.1.241.1.2.11.1.9" instance="ptpbaseClockPortAssociateEntry" alias="ptpCloPorAssPacRece" type="counter64"/>
+      <mibObj oid=".1.3.6.1.2.1.241.1.2.11.1.10" instance="ptpbaseClockPortAssociateEntry" alias="ptpCloPorAssoInErro" type="counter64"/>
+      <mibObj oid=".1.3.6.1.2.1.241.1.2.11.1.11" instance="ptpbaseClockPortAssociateEntry" alias="ptpCloPorAssOutErro" type="counter64"/>
+   </group>
+
+   <systemDef name="ptpBase">
+      <sysoidMask>.1.3.6.1.4.1.</sysoidMask>
```
Comment (indigo423) [inline opennms-base-assembly/src/main/filtered/etc/datacollection/ptpbase.xml:200]: I'm not sure if we want to try fetching these metrics from any MIB2-compatible SNMP agent on this planet. What about binding it to an `SNMP-PTP` service instead? It allows you to keep the systemDef global and gives users some control over where they want to try to fetch these metrics from. Maybe we can add an SNMP-PTP detector for it?

File: opennms-base-assembly/src/main/filtered/etc/poller-configuration.xml (hunk 1 above)
Comment (christianpape) [reply opennms-base-assembly/src/main/filtered/etc/poller-configuration.xml]: done.

File: opennms-base-assembly/src/main/filtered/etc/poller-configuration.xml (hunk 2 above)
Comment (christianpape) [reply opennms-base-assembly/src/main/filtered/etc/poller-configuration.xml:216]: done.

File: opennms-base-assembly/src/main/filtered/etc/datacollection/ptpbase.xml (hunk 3 above)
Comment (christianpape) [reply opennms-base-assembly/src/main/filtered/etc/datacollection/ptpbase.xml:58]: Not for SNMP resourceTypes. These are all still in the files under etc/datacollection/.

File: opennms-base-assembly/src/main/filtered/etc/datacollection/ptpbase.xml (hunk 4 above)
Comment (christianpape) [reply opennms-base-assembly/src/main/filtered/etc/datacollection/ptpbase.xml:200]: Enabled that for Arista devices for now.

File: opennms-base-assembly/src/main/filtered/etc/poller-configuration.xml [hunk 5]
```diff
@@ -213,6 +213,12 @@
          <parameter key="rrd-base-name" value="pop3"/>
          <parameter key="ds-name" value="pop3"/>
       </service>
+      <service name="PTP" interval="300000" user-defined="false" status="on">
+         <pattern><![CDATA[^PTP-(?<clockPort>.*)-(?<portState>.*)$]]></pattern>
```
Comment (indigo423) [inline opennms-base-assembly/src/main/filtered/etc/poller-configuration.xml:217]:
```suggestion
         <pattern><![CDATA[^PTP-.*$]]></pattern>
```

Comment (indigo423) [reply opennms-base-assembly/src/main/filtered/etc/poller-configuration.xml:217]: Allow a user to create any service name with PTP-* and apply the service config section to it. Remove the regexp group extraction for configs.

File: opennms-base-assembly/src/main/filtered/etc/poller-configuration.xml [hunk 6]
```diff
@@ -213,6 +213,12 @@
...
+         <parameter key="rrd-repository" value="${install.share.dir}/rrd/response" />
```
Comment (indigo423) [inline opennms-base-assembly/src/main/filtered/etc/poller-configuration.xml:218]:
```suggestion

```

File: opennms-base-assembly/src/main/filtered/etc/poller-configuration.xml [hunk 7]
```diff
@@ -213,6 +213,12 @@
...
+         <parameter key="clock-port" value="${pattern:clockPort|requisition:ptp:clockPort|default}" />
```
Comment (indigo423) [inline opennms-base-assembly/src/main/filtered/etc/poller-configuration.xml:219]:
```suggestion
         <parameter key="clock-port" value="${requisition:ptp:clockPort|default}" />
```

File: opennms-base-assembly/src/main/filtered/etc/poller-configuration.xml [hunk 8]
```diff
@@ -213,6 +213,12 @@
...
+         <parameter key="port-state" value="${pattern:portState|requisition:ptp:portState|default}" />
```
Comment (indigo423) [inline opennms-base-assembly/src/main/filtered/etc/poller-configuration.xml:220]:
```suggestion
         <parameter key="port-state" value="${requisition:ptp:portState|default}" />
```

**Review by indigo423 — COMMENTED** [review]
That's what I mean.

**Review by indigo423 — DISMISSED** [review]
LGTM!

**Review by indigo423 — APPROVED** [review]
Tears of joy! 🎊🌈🎉

**Review by fooker — APPROVED** [review]

**Pull request merged**